core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -101,6 +104,11 @@ private[spark] class ExecutorAllocationManager(
   private val executorIdleTimeoutS = conf.getTimeAsSeconds(
     "spark.dynamicAllocation.executorIdleTimeout", "60s")
 
+  private val cachedExecutorTimeoutS = conf.getTimeAsSeconds(
+    "spark.dynamicAllocation.cachedExecutorIdleTimeout", s"${Long.MaxValue / 1000}s")
+
+  val executorsWithCachedBlocks = new mutable.HashSet[String]()
```

Comment:
also, we never remove from this set. Over time this will build up if this is a long running application